tensorizer/serialization.py

```diff
@@ -433,19 +458,21 @@ def read_tensors(
         except EOFError:
             return
 
-    @staticmethod
-    def _to_torch_parameter(arr: numpy.ndarray,
+    def _to_torch_parameter(self,
+                            arr: Union[numpy.ndarray, torch.nn.Parameter],
                             dtype: Optional[str] = None,
                             device=utils.get_device()) -> torch.nn.Parameter:
         """
         Convert a numpy array to a torch tensor on a device.
         """
         if dtype is not None and arr.dtype != "bool" and arr.dtype != dtype:
             arr = arr.astype(dtype)
+        if isinstance(arr, torch.nn.Parameter):
+            return arr.to(device)
```

Comment:
This turns the parameter back into a tensor (I do not know why PyTorch does this), but this removes its property of being a parameter object.

Reply:
It can be fixed by replacing this with:
```python
        if isinstance(arr, torch.nn.Parameter):
            arr.data = arr.to(device)
            if arr.grad is not None:
                arr.grad = arr.grad.to(device)
            return arr
```